e2e-tests/tests/service-per-pod/04-assert.yaml

```diff
@@ -30,6 +30,8 @@ apiVersion: ps.percona.com/v1alpha1
 kind: PerconaServerMySQL
 metadata:
   name: service-per-pod
+  finalizers:
+    - delete-mysql-pods-in-order
```

Comment:
I think we need to add a test where we actually check that it is working as intended.

Reply:
According to @hors , we should just add the finalizer to the existing test cluster config. Also on other operators, I don't see anything that tests the deletion (or I miss it somehow).

Reply:
@tplavcic yes, we need to have it but we do not have such tests for all other our operators. That is why I suggested doing it via separate PR and maybe you can help us to add it (or help with use-cases.)

Reply:
created a ticket here: https://jira.percona.com/browse/K8SPS-154